### PR TITLE
Update Core.php

### PR DESCRIPTION
**Pull request by gstesq**
When exporting JS using module assets folder, server returning MIME type 'text/plain' - needs 'text/javascript'

### CODE REVIEW

**Review by gstesq — COMMENTED** [review]
Comments?

**Comment by jakecastelli** [comment]
Hi Gray, I will try to review this a bit later if that is ok.

**Review by jakecastelli — APPROVED** [review]
This is valid! If you do not set the Content-Type then the JavaScript Module (import/ export) will not work.

File: engine/Core.php [hunk 1]
```diff
@@ -110,6 +114,9 @@ private function serve_child_module_asset($asset_path, $file_name) {
                         if (is_numeric($pos2)) {
                             $content_type = 'text/css';
                         }
+                        if (is_numeric($pos2)) {
```
Comment (jakecastelli) [inline engine/Core.php:117]: don't you need this line here `$pos2 = strpos($file_name, '.js');` ?

Comment (gstesq) [reply engine/Core.php:117]: Yeah - thought I'd added it. Cheers.

**Comment by gstesq** [comment]
Also, I had to add another check:`                    if ($content_type == 'text/html') {
                        $pos2 = strpos($file_name, '.js');
                        if (is_numeric($pos2)) {
                            $content_type = 'text/javascript';
                        }
                    }`

File: engine/Core.php [hunk 2]
```diff
@@ -105,12 +117,21 @@ private function serve_child_module_asset($asset_path, $file_name) {
...
                             $content_type = 'text/css';
                         }
-                        
+                         $pos2 = strpos($file_name, '.js');
+                        if (is_numeric($pos2)) {
+                            $content_type = 'text/javascript';
+                        }
+                    }
+                    
+                    if ($content_type == 'text/html') {
```
Comment (gstesq) [inline engine/Core.php:130]: Added to return 'text/javascript' instead of 'text/html' when using string literals.

File: engine/Core.php (hunk 1 above)
Comment (gstesq) [reply engine/Core.php:132]: Also, I had to add another check: if ($content_type == 'text/html') { $pos2 = strpos($file_name, '.js'); if (is_numeric($pos2)) { $content_type = 'text/javascript'; } }

When using string literals in the javascript it was returning html.

**Comment by jakecastelli** [comment]
Just a small suggestion Gary, you can condense them by saying `if ($content_type == 'text/plain' || $content_type == 'text/html')`

**Comment by gstesq** [comment]
> Just a small suggestion Gary, you can condense them by saying `if ($content_type == 'text/plain' || $content_type == 'text/html')`

Refactored and updated.

**Review by davidjconnelly — APPROVED** [review]
Great and thank you!  If there's anything else, please do tweak away and submit another pull request.  I'm currently uploading to a new host and my brain is functioning at about 1%.  So, just do what you need to do.  

Super grateful, as always.  Thanks!

**Pull request merged**